.github/actions/setup-maven-cache/action.yaml

```diff
@@ -74,6 +74,13 @@ runs:
           echo "MAVEN_CACHE_KEY=${KEY_PREFIX}-full-${HASH_FILES}" >> $GITHUB_ENV
           echo "MAVEN_CACHE_KEY_PREFIX=${KEY_PREFIX}-full" >> $GITHUB_ENV
         fi
+
+      # Check if PR contains a label to disable the cache
+    - name: Check if cache is disabled
+      if: ${{ !(startsWith(github.ref_name, 'stable/') || github.ref_name == 'main') }}
```

Comment:
❓ I assume this condition is to make sure the `is-cache-disabled` action is only called on PRs? I'd prefer this logic to be residing in the `is-cache-disabled` inside that action so it always gives the "right" result, also `on: scheduled` etc workflows.